src/app/features/data-policy/data-policy.page.ts

```diff
@@ -31,11 +32,14 @@ export class DataPolicyPage {
       .pipe(
         tap(event => {
           const postMessageEvent = event as MessageEvent;
-          if (postMessageEvent.data === 'iframe-on-load') {
-            this.iframeLoaded$.next(true);
-          }
-          if (postMessageEvent.data === 'iframeBackButtonClicked') {
-            this.navController.back();
+          const data = postMessageEvent.data as BubbleToIonicPostMessage;
+          switch (data) {
+            case BubbleToIonicPostMessage.IFRAME_ON_LOAD:
+              this.iframeLoaded$.next(true);
+              break;
+            case BubbleToIonicPostMessage.IFRAME_BACK_BUTTON_CLICKED:
+              this.navController.back();
+              break;
           }
         }),
         untilDestroyed(this)
```

Comment:
**ChatGPT response for**  `git show a9e4232dfd700d07b6a61ec73910927941b129ae src/app/features/data-policy/data-policy.page.ts`

1.  It's good to use enums instead of string literals to avoid typo errors.
2.  It's better to define the type of the message event data for better code clarity.
3.  Consider adding a default case to the switch statement for unexpected post message events.

Reply:
@tammyyang regarding ChatGPT response:

1. ignore
2. make sense, but currently we just pass string as a message so I would ignore for now.
3. agree, I missed `default: case`. Created asana card [[Code Quality] Consider adding a default case to the switch statement for unexpected post message events](https://app.asana.com/0/1201016280880500/1204028658078831/f) _Find all switch statements in capture-app and make sure it has default case._